compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -751,10 +751,18 @@ impl<'a> Parser<'a> {
                     tail.could_be_bare_literal = true;
                     if maybe_struct_name.is_ident() && can_be_struct_literal {
                         // Account for `if Example { a: one(), }.is_pos() {}`.
+                        // expand `before` so that we take care of module path such as:
+                        // `foo::Bar { ... } `
+                        // we expect to suggest `(foo::Bar { ... })` instead of `foo::(Bar { ... })`
+                        let sm = self.sess.source_map();
+                        let before = maybe_struct_name.span.shrink_to_lo();
```

Comment:
It's not a great way, but in the parsing phase, we don't have more context here?
any better solution?

Reply:
> we don't have more context here?

Probably not.... We'd need to pass in more span information here, which it seems like we don't have.

Reply:
It seems we need to get the spans of several previous tokens here, since the module path may contain multiple tokens, while `parser` keep `prev_token` and `token`.

Reply:
I think this approach is ok